### PR TITLE
[Serializer] Fallback looking for DiscriminatorMap on interfaces

### LINKED ISSUE #47219: [Serializer] DiscriminatorMap with multiple interfaces/abstract class
### Symfony version(s) affected

6.1

### Description

When using the serializer to normalize an object that implements an interface with a DisciminatorMap and also extends from an abstract class, the ObjectSerializer doesn't use the DiscriminatorMap and makes the denormalize process failed.
The abstract class seems to have some sort of prority over the interfaces.
I couldn't find this behavior documented anywhere.

### How to reproduce

I have created a repository to reproduce the problem easily: https://github.com/yurplan/serializer-bug-example with a composition example.

Here is a minimal example:
```php
#[DiscriminatorMap(
    typeProperty: 'type',
    mapping: [
        'cat' => Cat::class,
        'dog' => Dog::class,
    ],
)]
interface Animal {}

abstract class AbstractEntity {}

class Cat extends AbstractEntity implements Animal {}

class Dog extends AbstractEntity implements Animal {}

$serializer->deserialize($serializer->serialize(new Cat(), 'json'));

// Throws PHP Fatal error:  Uncaught Symfony\Component\Serializer\Exception\NotNormalizableValueException: Type property "type" not found for the abstract object "Animal". in vendor/symfony/serializer/Exception/NotNormalizableValueException.php:31
```

### Possible Solution

I could create an abstract classes implementing the interfaces and extending from the previous class.
I could also create a normalizer to handle this case (i have seen https://github.com/symfony/symfony/issues/47208#issuecomment-1207423431).
But I think this issue could be addressed in the package itself by iterating over the interface and abstract classes when looking for a DiscriminatorMap.
There will still be an issue when 2 DiscriminatorMaps are found with the same key...

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Caligone**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #47219
| License       | MIT
| Doc PR        | No

First time here
Trying to properly fix #47219 but I'm not sure about the impact of this changes
I am also not sure either if it should be a 6.2 or 6.1 based branch

- [x] Globally validate the behavior
- [x] Add tests

### CODE REVIEW

**Comment by derrabus** [comment]
Hello. Sorry nobody has picked up your PR yet. I have to ask you to provide a test for your change. This helps a reviewer understand the problem that you're attempting to fix. The test also makes sure, your bug is not re-introduced by accident.

**Comment by Caligone** [comment]
No worries
As requested, I added a test
I would be happy to receive feedback and make adjustments as needed

File: src/Symfony/Bridge/Doctrine/Middleware/Debug/Query.php [hunk 1]
```diff
@@ -43,7 +43,7 @@ public function stop(): void
         }
     }
 
-    public function setParam(string|int $param, null|string|int|float|bool &$variable, int $type): void
```
Comment (mtarld) [inline src/Symfony/Bridge/Doctrine/Middleware/Debug/Query.php]: I don't think this change is related to the Serializer :slightly_smiling_face:

File: src/Symfony/Bridge/Doctrine/Tests/Middleware/Debug/MiddlewareTest.php [hunk 2]
```diff
@@ -145,23 +145,31 @@ public function testWithParamBound(callable $executeMethod)
     {
         $this->init();
 
-        $product = 'product1';
-        $price = 12.5;
-        $stock = 5;
+        $sql = <<<EOT
```
Comment (mtarld) [inline src/Symfony/Bridge/Doctrine/Tests/Middleware/Debug/MiddlewareTest.php]: Same for these changes

File: src/Symfony/Component/Serializer/Tests/Fixtures/DummyMessageNumberThree.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Tests\Fixtures;
+
+use Symfony\Component\Serializer\Annotation\Groups;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
```
Comment (mtarld) [inline src/Symfony/Component/Serializer/Tests/Fixtures/DummyMessageNumberThree.php:15]: I think you can be the author here :wink:

File: src/Symfony/Component/Serializer/Tests/Fixtures/DummyMessageNumberThree.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Tests\Fixtures;
+
+use Symfony\Component\Serializer\Annotation\Groups;
```
Comment (mtarld) [inline src/Symfony/Component/Serializer/Tests/Fixtures/DummyMessageNumberThree.php]: This use is useless

**Review by mtarld — CHANGES_REQUESTED** [review]
Almost good!

**Comment by mtarld** [comment]
As it's an old bug, could you target the `5.4` branch instead?

File: src/Symfony/Component/Serializer/Tests/Fixtures/DummyMessageNumberThree.php (hunk 3 above)
Comment (stof) [reply src/Symfony/Component/Serializer/Tests/Fixtures/DummyMessageNumberThree.php:15]: generally, we don't care about maintaining `@author` on test fixtures. Anyway, to see who worked on a file, the git history is a more reliable tool.

**Comment by Caligone** [comment]
Thanks for the feedback !
I rebased and fixed last issues

**Review by mtarld — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Caligone.

**Pull request merged**